src/game_battlealgorithm.cpp

```diff
@@ -647,7 +647,7 @@ void Game_BattleAlgorithm::AlgorithmBase::Apply() {
 		int hp = GetAffectedHp();
 		int target_hp = target->GetHp();
 		target->ChangeHp(IsPositive() ? hp : -hp);
-		if (IsAbsorb()) {
+		if (IsAbsorb() && !IsPositive()) {
```

Comment:
This looks incorrect. `IsPositive()` is a flag that is set for healing skills, where "healing skill" is defined as a skill targeting the same party as the source.

```
this->healing =
    skill.scope == lcf::rpg::Skill::Scope_ally ||
    skill.scope == lcf::rpg::Skill::Scope_party ||
    skill.scope == lcf::rpg::Skill::Scope_self;
```

Reply:
This !IsPositive() check is a preliminary work for an upcoming PR which handles negative attribute rates for RPG Maker 2003. This should stop the user of an absorbing skill healing himself if the absorption of the target fails due to "negative damage". This is because the IsPositive() flag gets inverted if the "negative damage" situations occurs.

Reply:
because this is an enhancement not in RPG2k(3) better add a small comment here.

Reply:
If this is part of your implementation of negative attributes in a future PR, lets take it out of here and move it to there.

Reply:
The !IsPositive() checks on IsAbsorb() have been removed now.